bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/util/ShellyVersionDTO.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.shelly.internal.util;
+
+import java.util.Comparator;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * {@link ShellyVersionDTO} compares 2 version strings.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class ShellyVersionDTO implements Comparator<Object> {
```

Comment:
I disagree with a DTO class implementing Comparator and instead it should be either a lambda, an anonymous class, or a named nested class in this DTO.

The best suggestion I can give to to make VersionTokenizer implement `Comparable<VersionTokenizer>`

Reply:
How to do that? My Java skills are limited

Reply:
@cpmeister could you help here?

Reply:
Where is this class used as a comparator?

Reply:
I removed the comparator, ShellyBaseHandler directly calls .compare()